java/ql/test/query-tests/DoNotCallFinalize/Test.java

```diff
@@ -0,0 +1,28 @@
+public class Test {
+    void f() throws Throwable {
+        // NON_COMPLIANT
+        this.finalize(); // $ Alert
+    }
+
+    void f1() throws Throwable {
+        f(); // COMPLIANT
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        // COMPLIANT: If a subclass overrides `finalize()`
+        // it must invoke the superclass finalizer explicitly.
+        super.finalize();
+    }
+
+    // Overload of `finalize`
+    protected void finalize(String s) throws Throwable {
+        System.out.println(s);
+    }
```

Comment:
Add `COMPLIANT` comment for consistency?

Reply:
This was just meant to be code to set up the `COMPLIANT` case at line 23, so I don't think it also needs a compliant comment. Although, looking at line 23 now, I don't know why I put the compliant comment on the method rather than on the overload method _call_. Updated that in [e621f9f](https://github.com/github/codeql/pull/19075/commits/e621f9fd4903d60ca266e87afbd8f9969955bddc).